tools/Performance/DynamoPerformanceTests/ResultsComparer.cs

```diff
@@ -0,0 +1,373 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.VisualBasic.FileIO;
+
+namespace DynamoPerformanceTests
+{
+    /// <summary>
+    /// Class that compares two sets of performance results: 
+    /// the "base" results, and the "diff" results.
+    /// </summary>
+    public class ResultsComparer
+    {
+        private class BenchmarkResult
+        {
+            /// <summary>
+            /// Name of the benchmark method
+            /// </summary>
+            internal string Method { get; set; }
+
+            /// <summary>
+            /// Name of test graph
+            /// </summary>
+            internal string Graph { get; set; }
+
+            /// <summary>
+            /// Mean benchmark time
+            /// </summary>
+            internal double Mean { get; set; }
+
+            /// <summary>
+            /// Error in results
+            /// </summary>
+            internal double Error { get; set; }
+
+            /// <summary>
+            /// Standard deviation of results
+            /// </summary>
+            internal double StdDev { get; set; }
+        }
+
+        private class BenchmarkComparison
+        {
+            /// <summary>
+            /// Results of base benchmark
+            /// </summary>
+            internal BenchmarkResult Base;
+
+            /// <summary>
+            /// Results of diff benchmark
+            /// </summary>
+            internal BenchmarkResult Diff;
+
+            /// <summary>
+            /// Diff mean time as a percentage of base mean time
+            /// </summary>
+            internal double MeanDelta
+            {
+                get { return Math.Round(100 * Diff.Mean / Base.Mean, 2); }
```

Comment:
Doesn't this calculation need to be `100 * (Diff - Base)/Base`, i.e. if `Diff` is the latest benchmark result?

Reply:
Ok, so I was imaging a slightly different metric—not the "percent improvement" you're describing, but the `Diff` as a percentage of `Base` (in other words, "_it got x% faster_" vs. "_it takes x% as long as it used to_", respectively). Now that I've thought about it, your metric seems more helpful. I also think this is probably what @mjkkirschner meant when he suggested adding (+,-) in a previous comment. I've updated the metrics to reflect the  "_it got x% faster_" line of thought.